### PR TITLE
Feature/separate docker prod

### PR DESCRIPTION
**Pull request by Steven-Eardley**
It occurred to me that NIMS might not wish to have the system publicly available while in development. I've added nginx to a production docker-compose with basic auth.

user name: nims-test
password: zaigii5R

by using `.override.yml` and `-production.yml`, you bring it up in dev as normal with
`docker-compose up`

but in production we'll want
`docker-compose -f docker-compose.yml -f docker-compose-production.yml up -d`

### CODE REVIEW

**Comment by Steven-Eardley** [comment]
Do we also want to stop exposing fedora and solr to the internet? (by moving those port definitions into the dev file)

**Comment by anusharanganathan** [comment]
> Do we also want to stop exposing fedora and solr to the internet? (by moving those port definitions into the dev file)

You could. We don't need to expose these ports. Is it possible to override this. Or you could create a development override if needed.

File: docker-compose.yml [hunk 1]
```diff
@@ -67,8 +67,7 @@ services:
       - .env
       - .env.production
     volumes:
-      - app:/data/tmp/uploads
```
Comment (anusharanganathan) [inline docker-compose.yml]: Sorry I made some commits which need to be copied over to your override files. I resolved the conflict in the docker-compose.yml file, but the overrides also need this.

Comment (anusharanganathan) [reply docker-compose.yml]: We so need to mount a directory here, otherwise I cannot run imports

**Pull request closed without merging**